test/integration/good/code-gen/cpp.expected

```diff
@@ -20110,6 +20110,573 @@ stan::model::model_base& new_model(
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+    }
+    num_params_r__ = 0U;
+    
+    try {
+      num_params_r__ += nt * ((2 * (2 - 1)) / 2);
+      num_params_r__ += NS;
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+    }
+  }
+  template <bool propto__, bool jacobian__, typename VecR, typename VecI, stan::require_vector_like_t<VecR>* = nullptr, stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr>
+  inline stan::scalar_type_t<VecR> log_prob_impl(VecR& params_r__,
+                                                 VecI& params_i__,
+                                                 std::ostream* pstream__ = nullptr) const {
+    using T__ = stan::scalar_type_t<VecR>;
+    using local_scalar_t__ = T__;
+    T__ lp__(0.0);
+    stan::math::accumulator<T__> lp_accum__;
+    static const char* function__ = "param_constraint_model_namespace::log_prob";
+(void) function__;  // suppress unused var warning
+
+    stan::io::reader<local_scalar_t__> in__(params_r__, params_i__);
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+
+    
+    try {
+      std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>> L_Omega;
+      L_Omega = std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>>(nt, Eigen::Matrix<local_scalar_t__, -1, -1>(2, 2));
+      stan::math::fill(L_Omega, DUMMY_VAR__);
+      
+      std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>> L_Omega_in__;
+      L_Omega_in__ = std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>>(nt, Eigen::Matrix<local_scalar_t__, -1, 1>(
+        ((2 * (2 - 1)) / 2)));
+      stan::math::fill(L_Omega_in__, DUMMY_VAR__);
+      
+      current_statement__ = 1;
+      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
+        current_statement__ = 1;
+        assign(L_Omega_in__, cons_list(index_uni(sym1__), nil_index_list()),
+          in__.vector(((2 * (2 - 1)) / 2)), "assigning variable L_Omega_in__");
+      }
+      current_statement__ = 1;
+      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
+        current_statement__ = 1;
+        if (jacobian__) {
+          current_statement__ = 1;
+          assign(L_Omega, cons_list(index_uni(sym1__), nil_index_list()),
+            stan::math::cholesky_corr_constrain(L_Omega_in__[(sym1__ - 1)],
+              2, lp__), "assigning variable L_Omega");
+        } else {
+          current_statement__ = 1;
+          assign(L_Omega, cons_list(index_uni(sym1__), nil_index_list()),
+            stan::math::cholesky_corr_constrain(L_Omega_in__[(sym1__ - 1)], 2),
+            "assigning variable L_Omega");
+        }}
```

Comment:
Actually if you can nix  `L_Omega_in__` it would be wayy more efficient to use [`in__.cholesky_factor_corr_constrain()`](https://github.com/stan-dev/stan/blob/develop/src/stan/io/reader.hpp#L1184)  like

```suggestion
    if (jacobian__) {
      current_statement__ = 1;
      assign(L_Omega, nil_index_list(),
      in__.cholesky_factor_corr_constrain(((2 * (2 - 1)) / 2), lp__),
       "assigning variable L_Omega");
    } else {
      current_statement__ = 1;
      assign(L_Omega, nil_index_list(), 
        in__.cholesky_factor_corr_constrain(((2 * (2 - 1)) / 2)),
        "assigning variable L_Omega");
    }
```